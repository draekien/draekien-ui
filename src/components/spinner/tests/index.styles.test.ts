import {
  spinnerBarCss,
  spinnerCircleWrapperCss,
  spinnerDotCss,
} from '../index.styles';

describe('Spinner Bar Css', () => {
  test('should match snapshot', () => {
    expect(
      spinnerBarCss({ size: 'medium', color: 'primary' })
    ).toMatchSnapshot();
  });

  test('should match snapshot when small', () => {
    expect(
      spinnerBarCss({ size: 'small', color: 'secondary' })
    ).toMatchSnapshot();
  });

  test('should match snapshot when large', () => {
    expect(spinnerBarCss({ size: 'large' })).toMatchSnapshot();
  });
});

describe('Spinner Dot Css', () => {
  test('should match snapshot', () => {
    expect(spinnerDotCss({ size: 'medium' })).toMatchSnapshot();
  });

  test('should match snapshot', () => {
    expect(spinnerDotCss({ size: 'small' })).toMatchSnapshot();
  });

  test('should match snapshot', () => {
    expect(spinnerDotCss({ size: 'large' })).toMatchSnapshot();
  });
});

describe('Spinner Circle Css', () => {
  test('should match snapshot', () => {
    expect(
      spinnerCircleWrapperCss({ size: 'medium', color: 'primary' })
    ).toMatchSnapshot();
  });

  test('should match snapshot', () => {
    expect(
      spinnerCircleWrapperCss({ size: 'small', color: 'secondary' })
    ).toMatchSnapshot();
  });

  test('should match snapshot', () => {
    expect(spinnerCircleWrapperCss({ size: 'large' })).toMatchSnapshot();
  });
});
