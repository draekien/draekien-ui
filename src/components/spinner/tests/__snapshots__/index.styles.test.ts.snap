// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Bar Css should match snapshot 1`] = `
Object {
  "animation": "_EMO_animation-95jn4c_@keyframes animation-95jn4c{
0% {
  transform: scale(1);
}

20% {
  transform: scale(1, 2);
}

40% {
  transform: scale(1);
}}_EMO_ 1s ease-in-out infinite",
  "backgroundColor": "primary",
  "borderRadius": "md",
  "display": "inline-flex",
  "height": "1.125rem",
  "ml": "0.125rem",
  "width": "0.25rem",
}
`;

exports[`Spinner Bar Css should match snapshot when large 1`] = `
Object {
  "animation": "_EMO_animation-95jn4c_@keyframes animation-95jn4c{
0% {
  transform: scale(1);
}

20% {
  transform: scale(1, 2);
}

40% {
  transform: scale(1);
}}_EMO_ 1s ease-in-out infinite",
  "backgroundColor": "primary",
  "borderRadius": "md",
  "display": "inline-flex",
  "height": "1.625rem",
  "ml": "0.125rem",
  "width": "0.375rem",
}
`;

exports[`Spinner Bar Css should match snapshot when small 1`] = `
Object {
  "animation": "_EMO_animation-95jn4c_@keyframes animation-95jn4c{
0% {
  transform: scale(1);
}

20% {
  transform: scale(1, 2);
}

40% {
  transform: scale(1);
}}_EMO_ 1s ease-in-out infinite",
  "backgroundColor": "secondary",
  "borderRadius": "md",
  "display": "inline-flex",
  "height": "0.625rem",
  "ml": "0.125rem",
  "width": "0.125rem",
}
`;

exports[`Spinner Circle Css should match snapshot 1`] = `
Object {
  "color": "primary",
  "display": "inline-flex",
  "height": "1.5rem",
  "stroke": 4,
  "width": "1.5rem",
}
`;

exports[`Spinner Circle Css should match snapshot 2`] = `
Object {
  "color": "secondary",
  "display": "inline-flex",
  "height": "1rem",
  "stroke": 4,
  "width": "1rem",
}
`;

exports[`Spinner Circle Css should match snapshot 3`] = `
Object {
  "color": "primary",
  "display": "inline-flex",
  "height": "2rem",
  "stroke": 4,
  "width": "2rem",
}
`;

exports[`Spinner Dot Css should match snapshot 1`] = `
Object {
  ":nth-of-type(1)": Object {
    "backgroundColor": "primary",
  },
  ":nth-of-type(2)": Object {
    "backgroundColor": "primary-hover",
  },
  ":nth-of-type(3)": Object {
    "backgroundColor": "primary-hover",
  },
  "animation": "_EMO_animation-178py0y_@keyframes animation-178py0y{
0% {
  transform: translateY(0);
}

20% {
  transform: translateY(-75%);
}

40% {
  transform: translateY(0);
}}_EMO_ 1s ease-in-out infinite",
  "borderRadius": "50%",
  "display": "inline-flex",
  "height": "0.75rem",
  "margin": "0.25rem",
  "width": "0.75rem",
}
`;

exports[`Spinner Dot Css should match snapshot 2`] = `
Object {
  ":nth-of-type(1)": Object {
    "backgroundColor": "primary",
  },
  ":nth-of-type(2)": Object {
    "backgroundColor": "primary-hover",
  },
  ":nth-of-type(3)": Object {
    "backgroundColor": "primary-hover",
  },
  "animation": "_EMO_animation-178py0y_@keyframes animation-178py0y{
0% {
  transform: translateY(0);
}

20% {
  transform: translateY(-75%);
}

40% {
  transform: translateY(0);
}}_EMO_ 1s ease-in-out infinite",
  "borderRadius": "50%",
  "display": "inline-flex",
  "height": "0.5rem",
  "margin": "0.25rem",
  "width": "0.5rem",
}
`;

exports[`Spinner Dot Css should match snapshot 3`] = `
Object {
  ":nth-of-type(1)": Object {
    "backgroundColor": "primary",
  },
  ":nth-of-type(2)": Object {
    "backgroundColor": "primary-hover",
  },
  ":nth-of-type(3)": Object {
    "backgroundColor": "primary-hover",
  },
  "animation": "_EMO_animation-178py0y_@keyframes animation-178py0y{
0% {
  transform: translateY(0);
}

20% {
  transform: translateY(-75%);
}

40% {
  transform: translateY(0);
}}_EMO_ 1s ease-in-out infinite",
  "borderRadius": "50%",
  "display": "inline-flex",
  "height": "1rem",
  "margin": "0.25rem",
  "width": "1rem",
}
`;
